tests/unit_test/chat/user_media_test.py

```
@@ -428,3 +430,123 @@ def test_save_whatsapp_media_content_meta_failure(mock_get):
+    mock_action_instance.save.assert_called_once()
+    mock_prompt_action_instance.save.assert_called_once()
+    mock_rule_instance.save.assert_called_once()
+
+    mock_logger.exception.assert_not_called()
+
+@patch("kairon.shared.chat.user_media.logger")
+def test_add_media_extraction_flow_if_not_exist_exception(mock_logger):
+    bot = "test_bot"
+
+    with patch("kairon.shared.chat.user_media.Actions.objects", side_effect=Exception("Database error")):
+        with pytest.raises(AppException, match="Failed to add media extraction flow: Database error"):
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)